cdap-watchdog/src/test/java/co/cask/cdap/logging/appender/kafka/TestKafkaLogging.java

```diff
@@ -84,4 +102,62 @@ public void testGetPrev() throws Exception {
   }
 
   // Note: LogReader.getLog is tested in LogSaverTest for distributed mode
+
+  @Test
+  public void testPartitionKey() throws Exception {
+    CConfiguration cConf = KAFKA_TESTER.getCConf();
+    // set kafka partition key tp application
+    cConf.set(LoggingConfiguration.KAFKA_PARTITION_KEY, "application");
+
+    Logger logger = LoggerFactory.getLogger("TestKafkaLogging");
+    LoggingContext loggingContext = new FlowletLoggingContext("TKL_NS_2", "APP_2", "FLOW_2", "FLOWLET_2",
+                                                              "RUN2", "INSTANCE2");
```

Comment:
you are just publishing from one flow context ? you need to publish from two program context (from same app) to make sure they are reaching the same partition right ?